### ISSUE TITLE
Rule to enforce space inside array reference?

### ISSUE THREAD

**Issue by janez-svetin** [palantir/tslint#3909]
Is there a lint rule which can report on space inside array brackets? My example:
```
// bad
return decodeURIComponent(result[ 1 ]);
//good
return decodeURIComponent(result[1]);
```
Or is there any other rule which can enforce this? Thanks

**Comment by janez-svetin** [comment]
For now I have written a simple custom rule which uses regex on the node text:
```ts
...
function walk(ctx) {
  return ts.forEachChild(ctx.sourceFile, function cb(node) {
    if (node.kind === ts.SyntaxKind.ElementAccessExpression){
      var nodeText = node.getText();
      // invalid foo[ bar]
      // invalid foo[bar ]
      // invalid foo[ bar ]
      // valid foo[bar]
      if (/\w+\[(\s[^\]]+|[^\]]*\s)]/g.test(nodeText)) {
        ctx.addFailure(node.getStart(), 0, Rule.FAILURE_STRING);
      }
    }
    return ts.forEachChild(node, cb);
  });
}
```

Any recommendations for improvements are highly appreciated.
